lib/langchain/assistants/assistant.rb

```diff
@@ -125,13 +125,12 @@ def submit_tool_output(tool_call_id:, output:)
     # Call to the LLM#chat() method
     #
     # @return [Langchain::LLM::BaseResponse] The LLM response object
-    def chat_with_llm
+    def chat_with_llm(tool_choice: "auto")
       params = {messages: thread.openai_messages}
 
       if tools.any?
         params[:tools] = tools.map(&:to_openai_tools).flatten
-        # TODO: Not sure that tool_choice should always be "auto"; Maybe we can let the user toggle it.
-        params[:tool_choice] = "auto"
+        params[:tool_choice] = tool_choice unless tool_choice == "none"
```

Comment:
I don't quite understand, what exactly does this fix?

Reply:
Intention (no current requirement) is to allow running just a specific tool, and adding the param to `chat_with_llm` 'fixed' testing of different scenarios with different stubs (I think).
It may not work as-is, however, and should have a separate test for running a single tool and maybe an example in the readme.

Reply:
@mattlindsey The rule of thumb is that if you need to change implementation in order to make the tests pass it's either 1) the implementation is wrong or 2) your tests are poorly written.